### PR TITLE
Display whitespace

### LINKED ISSUE #877: Display whitespace
I was making some temporary screenshots from my editor while out of wifi range and discovered a starling difference. My editor shows white space, including vertical lines.

Generally you might think that's not super useful, but it is VERY useful when dealing with JSON, and writing say a JSON Schema.

It would be great if such an option could be added.

<3

### PR DESCRIPTION
**Pull request by baktiaditya**
Closes #877

Screen Shot 2019-10-08 at 06 48 14

### CODE REVIEW

File: components/Carbon.js [hunk 1]
```diff
@@ -377,6 +379,16 @@ function useHighlightLoader() {
   }, [])
 }
 
+let showInvisibleLoaded = false
+function useShowInvisibleLoader() {
```
Comment (mfix22) [inline components/Carbon.js]: Let's adjust this to use `import()` instead, and only call `import()` if `whiteSpace` is true

Comment (mfix22) [reply components/Carbon.js]: There is probably no need to check if it is already loaded in this case

File: components/Settings.js [hunk 2]
```diff
@@ -95,6 +96,11 @@ function WindowSettings({
           />
         </div>
       )}
+      <Toggle
+        label="Whitespace"
```
Comment (mfix22) [inline components/Settings.js]: You can leave the variable as whitespace, but can you change the `label` to `"Hidden characters"`?

File: components/Carbon.js (hunk 1 above)
Comment (baktiaditya) [reply components/Carbon.js]: Sorry @mfix22 i don't know how to do a dynamic import based on boolean value 😢

Comment (baktiaditya) [reply components/Carbon.js]: @mfix22 sorry, i don't know how to do dynamic import properly 😞

Comment (baktiaditya) [reply components/Carbon.js]: many attempts failed

**Comment by mfix22** [comment]
@baktiaditya can you rebase it on `master` so that only your changes are present in this PR? Sorry about all the hassle! I don't have time to do it myself at the moment

**Comment by mfix22** [comment]
@baktiaditya what if instead of using cm-show-invisibles, we just replaced the whitespace characters in `this.props.children`? 

Something like: 
```js
this.props.children
  .replace(/(\r\n|\r|\n)/g, '¬\n')
  .replace(/ /g, '·')
```

Thoughts?

**Comment by mfix22** [comment]
@all-contributors add @baktiaditya for code

**Pull request merged**